packages/smooth_app/lib/generic_lib/loading_dialog.dart

```diff
@@ -72,8 +72,11 @@ class LoadingDialog<T> {
       return;
     }
     _popEd = true;
-    // Here we use the root navigator so that we can pop dialog while using multiple navigators.
-    Navigator.of(context, rootNavigator: true).pop(value);
+    // To avoid returning before the alertDialog is build
+    SchedulerBinding.instance?.addPostFrameCallback((_) {
```

Comment:
I've always wanted to know how to deal with the "run a Future" and "display a dialog at the same time", especially when the "Future" is very fast.
Maybe you've found a correct solution, so congratulations!

Reply:
Thanks @monsieurtanuki, I reverted the changes to the other file :)